tests/h/util/uri_test.py

```diff
@@ -4,183 +4,302 @@
+            # Query: preserve relative ordering of multiple params with the same name
+            (
+                "http://example.com?b=2&b=3&b=1&a=1",
+                "httpx://example.com?a=1&b=2&b=3&b=1",
+            ),
+            ("http://example.com?b=&b=3&b=1&a=1", "httpx://example.com?a=1&b=&b=3&b=1"),
+        ),
+    )
+    def test_it_sorts_params(self, url_in, url_out):
+        assert uri.normalize(url_in) == url_out
+
+    @pytest.mark.parametrize(
+        "url_in,url_out",
+        (
+            # Query: remove parameters known to be irrelevant for document identity
+            ("http://example.com?utm_source=abcde", "httpx://example.com"),
+            ("http://example.com?utm_medium=abcde", "httpx://example.com"),
+            ("http://example.com?utm_term=abcde", "httpx://example.com"),
+            ("http://example.com?utm_content=abcde", "httpx://example.com"),
+            ("http://example.com?utm_campaign=abcde", "httpx://example.com"),
+            (
+                "http://example.com?utm_source=abcde&utm_medium=wibble",
+                "httpx://example.com",
+            ),
+            ("http://example.com?a=1&utm_term=foo", "httpx://example.com?a=1"),
+            ("http://example.com?a=1&utm_term=foo&b=2", "httpx://example.com?a=1&b=2"),
+            ("http://example.com?WT.mc_id=TWT_NatureNews", "httpx://example.com"),
+            ("http://example.com?WT.foo=bar", "httpx://example.com"),
+            (
+                "http://example.com?a=1&x-amz-security-token=abcde",
+                "httpx://example.com?a=1",
+            ),
+            (
+                "http://example.com?a=1&X-Amz-Security-Token=abcde",
+                "httpx://example.com?a=1",
+            ),
+            # but don't be over-eager and remove close matches
+            (
+                "http://example.com?gclid_foo=abcde",
+                "httpx://example.com?gclid_foo=abcde",
+            ),
+            (
+                "http://example.com?bar_gclid=abcde",
+                "httpx://example.com?bar_gclid=abcde",
+            ),
+            ("http://example.com?WT=abcde", "httpx://example.com?WT=abcde"),
+            (
+                "http://example.com?x-amz-security-token-foo=abcde",
+                "httpx://example.com?x-amz-security-token-foo=abcde",
+            ),
+        ),
+    )
+    def test_it_black_lists_invalid_params(self, url_in, url_out):
+        assert uri.normalize(url_in) == url_out
+
+
+class TestURIOrigin:
+    @pytest.mark.parametrize(
+        "url_in,url_out",
+        [
+            ("https://example.com", "https://example.com"),
+            ("https://example.com/foo?bar=baz#frag", "https://example.com"),
+            ("http://localhost:3000/foo", "http://localhost:3000"),
+            ("HTTP://LOCALHOST:3000/foo", "http://LOCALHOST:3000"),
+        ],
+    )
+    def test_it(self, url_in, url_out):
+        assert uri.origin(url_in) == url_out
```

Comment:
This guy was made into an object style test mostly for consistency with the others.